src/versionize/src/primitives.rs

```diff
@@ -62,6 +68,114 @@ impl_versionize!(f64);
 impl_versionize!(char);
 impl_versionize!(String);
 
+macro_rules! impl_atomic_versionize {
+    ($ty:ident) => {
+        impl Versionize for $ty {
+            #[inline]
+            fn serialize<W: std::io::Write>(
+                &self,
+                writer: &mut W,
+                _version_map: &VersionMap,
+                _version: u16,
+            ) -> VersionizeResult<()> {
+                bincode::serialize_into(writer, &self.load(Ordering::SeqCst))
+                    .map_err(|ref err| Error::Serialize(format!("{:?}", err)))?;
+                Ok(())
+            }
+
+            #[inline]
+            fn deserialize<R: std::io::Read>(
+                mut reader: &mut R,
+                _version_map: &VersionMap,
+                _version: u16,
+            ) -> VersionizeResult<Self>
+            where
+                Self: Sized,
+            {
+                Ok($ty::new(bincode::deserialize_from(&mut reader).map_err(
+                    |ref err| Error::Deserialize(format!("{:?}", err)),
+                )?))
+            }
+
+            // Not used.
+            fn version() -> u16 {
+                1
+            }
+        }
+    };
+}
+
+impl_atomic_versionize!(AtomicU8);
+impl_atomic_versionize!(AtomicU16);
+impl_atomic_versionize!(AtomicU32);
+impl_atomic_versionize!(AtomicU64);
+impl_atomic_versionize!(AtomicI8);
+impl_atomic_versionize!(AtomicI16);
+impl_atomic_versionize!(AtomicI32);
+impl_atomic_versionize!(AtomicI64);
+impl_atomic_versionize!(AtomicIsize);
+impl_atomic_versionize!(AtomicUsize);
+impl_atomic_versionize!(AtomicBool);
+
+impl Versionize for GuestAddress {
```

Comment:
I don't think `GuestAddress` is a primitive, see https://github.com/firecracker-microvm/firecracker/pull/1743#discussion_r402360347 for some more rambling on this subject.

Reply:
I totally agree for it not being a primitive, but this isn't a one way door we are taking here. Once the serialization stuff gets ironed out in rust-vmm we will definitely switch to that solution.